src/test/java/com/rackspace/salus/monitor_management/services/MonitorManagementPolicyTest.java

```diff
@@ -966,6 +967,41 @@ public void testHandleMonitorPolicyEvent_removePolicy_butKeepMonitor() {
     assertTrue(monitorRepository.findByTenantIdAndPolicyId(tenantId, policyId).isEmpty());
   }
 
+  /**
+   * Receive a remove monitor policy event and process it for a tenant
+   * that was previously using the policy and its entry in monitor_policies having
+   * null as monitor id
+   *
+   */
+  @Test
+  public void testHandleMonitorPolicyEvent_removePolicy_null_monitor_id() {
+    String tenantId = RandomStringUtils.randomAlphanumeric(10);
+    UUID policyId = UUID.randomUUID();
+    UUID policyMonitorId = currentMonitor.getId();
+
+    // store a monitor for the tenant that is tied to the policy in the event
+    Monitor clonedMonitor = createMonitorForPolicyForTenant(tenantId, policyId);
+
+    List<MonitorPolicyDTO> list = List.of((MonitorPolicyDTO) new MonitorPolicyDTO()
+        .setMonitorId(null)
+        .setId(policyId));
+
+    doReturn(list).when(policyApi).getEffectiveMonitorPoliciesForTenant(anyString(), anyBoolean());
```

Comment:
Mockito recommends not using `doReturn` except in rare cases

https://javadoc.io/static/org.mockito/mockito-core/3.5.15/org/mockito/Mockito.html#doReturn-java.lang.Object-
```suggestion
    when(policyApi.getEffectiveMonitorPoliciesForTenant(anyString(), anyBoolean())
      .thenReturn(list);
```